### PR TITLE
refactor(*) response encoding

### PR DESCRIPTION
**Pull request by gszr**
Response struct has response and error fields; `bridge.Ask` was updated
to handle the 3 possible scenarios in PDK responses:
- successful call with no result (nil, nil)
- successful call with result (res, nil)
- failed call with an error message (nil, err)

---

Fix issue caught in tests - missing arguments.

### CODE REVIEW

File: bridge/bridge.go [hunk 1]
```diff
@@ -19,10 +24,18 @@ func (b PdkBridge) Ask(method string, args ...interface{}) (string, error) {
 	} else {
 		call := method + ":" + string(argsJson)
 		b.ch <- call
-		if reply := <-b.ch; reply == "null" {
-			return "", errors.New("null response")
+		reply := <-b.ch
+		res := Response{}
+
+		if err := json.Unmarshal([]byte(reply), &res); err != nil {
```
Comment (javierguerragiraldez) [inline bridge/bridge.go]: now that it's always JSON on both call and response, it's getting more important to test msgpack...

File: bridge/bridge.go [hunk 2]
```diff
@@ -9,6 +9,11 @@ type PdkBridge struct {
 	ch chan string
 }
 
+type Response struct {
+	Res *string
+	Err *string
```
Comment (javierguerragiraldez) [inline bridge/bridge.go]: did you try without the pointers?  strings are already a reference type

File: bridge/bridge.go (hunk 1 above)
Comment (gszr) [reply bridge/bridge.go]: agreed! we will be looking at the serialization format next

File: bridge/bridge.go (hunk 2 above)
Comment (gszr) [reply bridge/bridge.go]: Done!

**Pull request closed without merging**